src/signTx.h

```diff
@@ -33,32 +35,48 @@ typedef enum {
 	SIGN_STAGE_BODY_CERTIFICATES_POOL_SUBMACHINE = 32, // pool registration certificate sub-machine
 	SIGN_STAGE_BODY_WITHDRAWALS = 33,
 	SIGN_STAGE_BODY_VALIDITY_INTERVAL = 34,
-	SIGN_STAGE_CONFIRM = 35,
-	SIGN_STAGE_WITNESSES = 36,
+	SIGN_STAGE_BODY_MINT = 35,
+	SIGN_STAGE_BODY_MINT_SUBMACHINE = 36,
+	SIGN_STAGE_CONFIRM = 37,
+	SIGN_STAGE_WITNESSES = 38,
 } sign_tx_stage_t;
 
 enum {
-	SIGN_MAX_INPUTS = 1000,
-	SIGN_MAX_OUTPUTS = 1000,
-	SIGN_MAX_CERTIFICATES = 1000,
-	SIGN_MAX_REWARD_WITHDRAWALS = 1000
+	SIGN_MAX_INPUTS = UINT16_MAX,
+	SIGN_MAX_OUTPUTS = UINT16_MAX,
+	SIGN_MAX_CERTIFICATES = UINT16_MAX,
+	SIGN_MAX_REWARD_WITHDRAWALS = UINT16_MAX,
+	SIGN_MAX_WITNESSES = SIGN_MAX_INPUTS + SIGN_MAX_OUTPUTS + SIGN_MAX_CERTIFICATES + SIGN_MAX_REWARD_WITHDRAWALS,
 };
 
 typedef struct {
 	// significantly affects restrictions on the tx
-	sign_tx_usecase_t signTxUsecase;
+	sign_tx_signingmode_t txSigningMode;
 
 	uint8_t networkId; // part of Shelley address
 	uint32_t protocolMagic; // part of Byron address
 } common_tx_data_t;
 
+typedef struct {
+	stake_credential_type_t type;
+	union {
+		bip44_path_t pathSpec; // interpretation depends on type // TODO rename to keyPath?
+		uint8_t scriptHash[SCRIPT_HASH_LENGTH];
+	};
+} stake_credential_t;
+
 typedef struct {
 	certificate_type_t type;
-	bip44_path_t pathSpec; // interpretation depends on type // TODO rename to keyPath?
 
-	// only for specific types
-	uint8_t poolKeyHash[POOL_KEY_HASH_LENGTH];
-	uint64_t epoch;
+	union {
+		stake_credential_t stakeCredential;
+		struct {
+			bip44_path_t poolIdPath;
+			uint64_t epoch;
+			uint8_t poolKeyHash[POOL_KEY_HASH_LENGTH];
+		};
+	};
+	
```

Comment:
We need to store pool id for stake_delegation. Right now, it goes into poolKeyHash, so it overlaps with stakeCredential within this union.

Reply:
True. Looking through the code, only the stakeCredential and the poolIdPath is really suitable to be in a union.